cpp/src/backends/handler/base_handler.cc

```diff
@@ -63,6 +63,10 @@ std::shared_ptr<torch::Device> BaseHandler::GetTorchDevice(
     return std::make_shared<torch::Device>(torch::kCPU);
   }
 
+#if defined(__HIPCC__) || (defined(__clang__) && defined(__HIP__)) || defined(__HIPCC_RTC__)
+  return std::make_shared<torch::Device>(torch::kHIP,
```

Comment:
Why are you using HIP for the device here instead of CUDA?  ROCm PyTorch masquerades as the CUDA device.  Though a HIP device does exist as a dispatch key, no kernels are registered to it.